portfolio/src/main/java/com/google/sps/servlets/CommentsServlet.java

```diff
@@ -31,42 +31,54 @@
+  private static final String MAX_COMMENTS_QUERY_STRING = "max-comments";
+  private static final String COMMENT_FORM_SUBMIT = "comment-submit";
+  private static final String MAX_FORM_SUBMIT = "max-comment-submit";
+  private static final int MAX_COMMENT_DEFAULT = 5;
+  int maxCommentsToDisplay;
   DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
   CommentDatabase commentDatabase = new CommentDatabase(datastore);
 
+  @Override
+   public void init() {
+     maxCommentsToDisplay = MAX_COMMENT_DEFAULT;
+   }
+
   /* responds with a JSON string containing authors and comments*/
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // Query datastore for all comments
-    int maxComments = MAX_COMMENT_DEFAULT;
-    try {
-        maxComments = getMaxCommentsToDisplay(request);
-    } catch (IllegalArgumentException e) {
-        maxComments = MAX_COMMENT_DEFAULT;
-        System.out.println(e.getMessage());
-    } 
-    ArrayList<Comment> comments = commentDatabase.getComments(maxComments);
+    ArrayList<Comment> comments = commentDatabase.getComments(maxCommentsToDisplay);
     // Convert the ArrayLists to JSON
     Gson gson = new Gson();
     String commentsJson = gson.toJson(comments);
     // Send the JSON as the response
     response.setContentType(JSON_CONTENT_TYPE);
     response.getWriter().println(commentsJson);
   }
-
-  /* when the submit button on the values page is hit, doPost requests the text in the author and commment field and stores them*/ 
+  
+  /* responds to comments and max comment limit set by user */
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // Get the input from the comments form
-    try {
-      String authorText = getParameter(request, commentDatabase.AUTHOR_QUERY_STRING);
-      String valueText = getParameter(request, commentDatabase.VALUE_QUERY_STRING);
-      Comment comment = new Comment(authorText, valueText);
-      commentDatabase.putCommentInDatabase(comment);   
-    } catch (RuntimeException e){ 
+    // if max comments submit button clicked, get input from max comments form 
+    if (getParameter(request,  MAX_FORM_SUBMIT) != "") {
+      try {
+        setMaxCommentsToDisplay(request);
+      } catch (IllegalArgumentException e) {
+        maxCommentsToDisplay = MAX_COMMENT_DEFAULT;
+        System.out.println(e.getMessage());
+      } 
+    }
+    // if comment submit button clicked, get author and value from comments form
+    if (getParameter(request, COMMENT_FORM_SUBMIT) != "") {
+      try {
+        String authorText = getParameter(request, commentDatabase.AUTHOR_QUERY_STRING);
+        String valueText = getParameter(request, commentDatabase.VALUE_QUERY_STRING);
+        Comment comment = new Comment(authorText, valueText);
+        commentDatabase.putCommentInDatabase(comment);   
+      } catch (RuntimeException e) { 
```

Comment:
These probably belong in their own methods now.

Reply:
I agree

Reply:
Good idea, done.